### ISSUE TITLE
LiquibaseDatabase should forward additional changeLogParameters

### ISSUE THREAD

**Issue by crazy2398** [jOOQ/jOOQ#17593]
### Use case

The code generation by LiquibaseDatabase doest work when I was tring to use some liquibase changeLog like this:
```
<createTable tableName="book">
    <column name="id" type="${key_type}">
        <constraints nullable="false" primaryKey="true" primaryKeyName="pk_book"/>
    </column>
    ......
</createTable>
```
becase the custom parameter place holder "${key_type}" is not be replaced by the string which I had set in the generator database properties like this (By Gradle):
```
jooq {
      configuration {
            generator {
                database {
                      name="org.jooq.meta.extensions.liquibase.LiquibaseDatabase"
                      properties {
                            ......
                            property {
                                  key = "changeLogParameters.key_type"
                                  value = "int8"
                            }
......
```
and so, I read the source and find it's effective only on the key = "changeLogParameters.contexts", [#9872](https://github.com/jOOQ/jOOQ/issues/9872).

in my case, the data type of the table‘s primary key has not been determined, our team is falling into the dispute of that. but with the custom parameter of liquibase, we can push other important works ahead. I believe the custom parameters of liquibase has other important usecase, so I hope JOOQ can support that.

### Possible solution

Liquibase class accepts a Map<String, String> for the external custom parameters，the methd name is setChangeLogParameter(), so, actually, I had write a class cpoied form LiquibaseDatabase, which is changed like this to implement what I want:
```
public class MyLiquibaseDatabase extends AbstractInterpretingDatabase {
    ......
    protected void export() throws Exception {
        String contexts = "";
        Map<String, String> changeLogParams = new HashMap<>();  // this line is the code I added
        ......
        for (Entry<Object, Object> entry : getProperties().entrySet()) {
        ......
            // [#9872] Some changeLogParameters can also be passed along
           else if (key.startsWith("changeLogParameters.")) {
                  String property = key.substring("changeLogParameters.".length()); 
                  if ("contexts".equals(property))
                        contexts = "" + entry.getValue();
                  else   // this line is the code I added
                    changeLogParams.put(property, Objects.toString(entry.getValue(), ""));// this line is the code I added
              }
        }
        ......
        Liquibase liquibase = new Liquibase(scripts, ra, database);
        changeLogParams.forEach(liquibase::setChangeLogParameter);   // this line is the code I added
        liquibase.update(contexts);
    }
}
```
this is my resolution for this issue. if you accepted this feature, I hope my code maybe could give you some ideas or helpful to reduce your coding burden.
At last, Thank you for JOOQ. It allows me to write my favorite SQL in my favorite Java freely.

### jOOQ Version

JOOQ CE 3.19.14

### Database product and version

Liquibase

### Java Version

oracle JDK 21

**Comment by lukaseder** [comment]
Thanks for your suggestion. I can see this as a valuable feature addition. Here's the documentation to the relevant Liquibase feature:

https://docs.liquibase.com/concepts/changelogs/property-substitution.html

Of course, you can also use any other means of propagating properties to Liquibase, including system properties, environment variables (according to their docs).

**Comment by lukaseder** [comment]
Implemented for jOOQ 3.20.0
